nmtwizard/preprocess/tu.py

```diff
@@ -196,7 +196,7 @@ def replace_tokens(self, start_idx, tok_num, new_tokens=None, part=0):
                 if start_idx < cur_length:
                     del cur_tokens[start_idx:end_idx]
             else:
-                new_tokens = list(map(pyonmttok.Token, new_tokens))
+                new_tokens = self.__tokenizer.deserialize_tokens(new_tokens)
```

Comment:
Maybe we should check that `new_tokens` does not already contain `Token` objects?